### ISSUE TITLE
Enable code splitting at the route level

### ISSUE THREAD

**Issue by seriouslysean** [cuttle-cards/cuttle#563]
### Feature Summary

Async load top level view components to take advantage of code splitting

### Detailed Description

Convert routes to something like this:
```
const routes = [
    {
        path: '/path',
        component: () => import('/path/to/component')
    }
]
```

**Comment by seriouslysean** [comment]
See https://github.com/cuttle-cards/cuttle/blob/main/src/router.js#L43

**Comment by Nishith-Savla** [comment]
Hey @seriouslysean, I would like to work on this. Can you please assign it to me?

**Comment by seriouslysean** [comment]
@Nishith-Savla  when working on this, 2 things to look out for:

- code splits properly and the bundle decreases in size for each route.
- The user experience is not impacted by longer load times.

I expect time to first byte will improve, but I don’t want that to be at the cost of navigation speeds when browsing from page to page.

Keeping in mind the critical path is Home/GameList -> Lobby -> Game.

**Comment by Nishith-Savla** [comment]
Yes will surely keep in check. Also, how do I check the bundle size for each route? Through the browser network captures?

**Comment by seriouslysean** [comment]
Browser capture is totally fine. Some sort of bundle analyzer or do a production build and check files on the hard drive.

**Comment by Nishith-Savla** [comment]
Ok will try.

**Comment by ar1701** [comment]
Hi @seriouslysean , I would to contribute into this. Can you please assign this to me ?

**Comment by IamPossible007** [comment]
HI @seriouslysean , is this issue resolved because in  https://github.com/cuttle-cards/cuttle/blob/main/src/router.js#L43
I find most of the routes have been configured as you wished , so if this issue is still open , I would like to contribute to it , this is my first time contributing to open source , I would like to start my journey , if you can help me , would be grateful.

**Comment by ar1701** [comment]
I am trying to create a pr, but this is coming
At least 1 approving review is required by reviewers with write access.

**Comment by Haviles04** [comment]
> I am trying to create a pr, but this is coming
> At least 1 approving review is required by reviewers with write access.
> @seriouslysean can u help me with this ?
> 

Your PR is created, it is waiting for review, and approval from the Core Team before it can be merged into the codebase
